asciidoc-confluence-publisher-client/src/test/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisherTest.java

```diff
@@ -68,6 +70,9 @@
     private static final String TEST_RESOURCES = "src/test/resources/org/sahli/asciidoc/confluence/publisher/client";
     private static final String SOME_CONFLUENCE_CONTENT_SHA256_HASH = "7a901829ba6a0b6f7f084ae4313bdb5d83bc2c4ea21b452ba7073c0b0c60faae";
 
+    ConfluencePublisherListener confluencePublisherListenerMock = mock(ConfluencePublisherListener.class);
```

Comment:
We try to avoid creating instance state in unit tests, instead we create and record mocks directly within the test method. Therefore, if you don't mind, please revert does changes again (also makes the pull request smaller ;-))

Reply:
@cstettler sorry didn't get that. Junit creates a new instance of test class for each new test method, so there is no need to copy paste these mocks in each and every test. Please checkout the examples provided in Mockito repository: https://github.com/mockito/mockito/blob/release/2.x/src/test/java/org/mockitousage/annotation/MockInjectionUsingConstructorTest.java
Of course, I can rollback the changes, but I don't see any point in copypasting test code. Test code is the same as production code and it needs to be maintainable... Is there any specific intention for test code duplication?

Reply:
I am aware of the Mockito injection support, and technically, your approach is of course working fine. The reason why we prefer the existing approach is readability of the individual tests: I want to be able to understand what the test is doing by only reading the test method. 

When using members in tests, it happens quickly to harm this obvious readability. Having a look at any of the tests in here and the utility method `confluencePublisher(...)`shows that: the test records the `confluenceRestClientMock` mock, but it is not visible in the test where this mock is used and what for, i.e. by the `ConfluencePublisher` instance created in `confluencePublisher(...)`. Because the mocks are accessible as members, it is no longer required to explicitly pass them into the utility methods (of course, it could still be done, but is often not).

I value readability over strict de-duplication (especially as in this case, there is no duplication of logic that might evolve separately and thus, get inconsistent). Of course that perspective is debatable.

Reply:
@cstettler ok, good. the intention is clear to me, although, of course, as you said it's debatable :) i will revert these changes